dhis-2/dhis-api/src/main/java/org/hisp/dhis/program/ChangeLogableEvent.java

```diff
@@ -32,6 +32,10 @@
 import java.util.Date;
 import org.locationtech.jts.geom.Geometry;
 
+/**
+ * Represents an event that can be logged in the change log system. Implementations of this
+ * interface should provide information about the properties that can be logged.
```

Comment:
> Implementations of this interface should provide information about the properties that can be logged.

How would implementations provide this information? I don't get what I would need to do.

Reply:
I removed the interface.
It was not clear because it was just an attempt at reducing duplicated code but in the wrong way.
Now every entity is passing to changeLog all the properties that makes sense for that entity and there is no NPE risk anymore.